move/sources/food.move

```diff
@@ -0,0 +1,161 @@
+
+
+    struct FoodToken has key {
+        /// Used to mint fungible assets.
+        fungible_asset_mint_ref: fungible_asset::MintRef,
+        /// Used to burn fungible assets.
+        fungible_asset_burn_ref: fungible_asset::BurnRef,
+    }
+
+    fun init_module(account: &signer) {
+        create_food_collection(account);
+
+        create_food_token(
+            account,
+            string::utf8(FOOD_DESCRIPTION),
+            string::utf8(FOOD_NAME),
+            string::utf8(FOOD_URI),
+            string::utf8(FOOD_NAME),
+            string::utf8(FOOD_NAME),
+            string::utf8(FOOD_URI),
+            string::utf8(PROJECT_URI),
+        );
+    }
+
+    /// Creates the food collection.
+    fun create_food_collection(creator: &signer) {
+        // Constructs the strings from the bytes.
+        let description = string::utf8(FOOD_COLLECTION_DESCRIPTION);
+        let name = string::utf8(FOOD_COLLECTION_NAME);
+        let uri = string::utf8(FOOD_COLLECTION_URI);
+
+        collection::create_unlimited_collection(
+            creator,
+            description,
+            name,
+            option::none(),
+            uri,
+        );
+    }
+
+    /// Creates the food token as fungible token.
+    fun create_food_token(
+        creator: &signer,
+        description: String,
+        name: String,
+        uri: String,
+        fungible_asset_name: String,
+        fungible_asset_symbol: String,
+        icon_uri: String,
+        project_uri: String,
+    ) {
+        let collection = string::utf8(FOOD_COLLECTION_NAME);
+        let constructor_ref = token::create_named_token(
+            creator,
+            collection,
+            description,
+            name,
+            option::none(),
+            uri,
+        );
+
+        let token_signer = object::generate_signer(&constructor_ref);
+
+        // Creates the fungible asset.
+        primary_fungible_store::create_primary_store_enabled_fungible_asset(
+            &constructor_ref,
+            option::none(),
+            fungible_asset_name,
+            fungible_asset_symbol,
+            0,
+            icon_uri,
+            project_uri,
+        );
+        let fungible_asset_mint_ref = fungible_asset::generate_mint_ref(&constructor_ref);
+        let fungible_asset_burn_ref = fungible_asset::generate_burn_ref(&constructor_ref);
+
+        // Publishes the FoodToken resource with the refs.
+        let food_token = FoodToken {
+            fungible_asset_mint_ref,
+            fungible_asset_burn_ref,
+        };
+        move_to(&token_signer, food_token);
+    }
+
+    public fun get_food_token_address(): address {
+        let food_token_address = token::create_token_address(
+            &@aptogotchi,
+            &string::utf8(FOOD_COLLECTION_NAME),
+            &string::utf8(FOOD_NAME),
+        );
+
+        food_token_address
+    }
+
+    public fun mint_food(user: &signer, amount: u64) acquires FoodToken {
```

Comment:
Anyone can call this function to bypass your pricing in aptogotchi::buy_food

Reply:
I thought about this but I don't know what is the best way to implement given this is a demo purpose. I don't want to introduce the friend concept (yet).